api/api.go

```diff
@@ -5,39 +5,59 @@ import (
 	"encoding/json"
 	"fmt"
 	"net/http"
+	"strings"
 )
 
 var ErrBadStatusCode = fmt.Errorf("bad status code")
 
 type Client struct {
-	BaseURL string
-	Token   string
+	BaseURL           string
+	Token             string
+	BasicAuthUser     string
+	BasicAuthPassword string
 }
 
 func NewClient(baseURL string, token string) Client {
-	return Client{
-		BaseURL: baseURL,
-		Token:   token,
+	auth := strings.SplitN(token, ":", 2)
```

Comment:
This is similar to the terraform provider: https://github.com/grafana/terraform-provider-grafana/blob/main/internal/provider/configure_clients.go#L204